tests/components/waterfurnace/conftest.py

```diff
@@ -0,0 +1,75 @@
+"""Fixtures for WaterFurnace integration tests."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, Mock, patch
+
+import pytest
+
+from homeassistant.components.waterfurnace.const import DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry, load_json_object_fixture
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock]:
+    """Mock setting up a config entry."""
+    with patch(
+        "homeassistant.components.waterfurnace.async_setup_entry", return_value=True
+    ) as mock_setup:
+        yield mock_setup
+
+
+@pytest.fixture
+def mock_waterfurnace_client() -> Generator[Mock]:
+    """Mock WaterFurnace client."""
+    with (
+        patch(
+            "homeassistant.components.waterfurnace.config_flow.WaterFurnace",
+            autospec=True,
+        ) as mock_client_class,
+        patch(
+            "homeassistant.components.waterfurnace.WaterFurnaceData",
+            new=mock_client_class,
+        ),
+    ):
+        mock_client = mock_client_class.return_value
+
+        # Mock the login method (blocking call)
+        mock_client.login = Mock()
+
+        # Mock the read method (blocking call) with fixture data
+        device_data = load_json_object_fixture("device_data.json", DOMAIN)
+
+        # Create a mock data object with attributes matching the fixture
+        mock_data = Mock()
+        for key, value in device_data.items():
+            setattr(mock_data, key, value)
```

Comment:
How does the object get created in the library? I would personally only opt to use JSON if the source data is also in JSON and you have an easy way to get an object like in the library

Reply:
The internal data object `WFReading` is initialized from the JSON returned by the service. I've updated this mock, and the fixture, to more closely reflect real data and have the `read` mock below return a `WFReading` object with the mock data instead.